packages/@aws-cdk-testing/framework-integ/test/core/test/integ.removal-policies.ts

```diff
@@ -0,0 +1,22 @@
+import { App, RemovalPolicies, Stack } from 'aws-cdk-lib';
+import * as dynamodb from 'aws-cdk-lib/aws-dynamodb';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as s3 from 'aws-cdk-lib/aws-s3';
+import * as integ from '@aws-cdk/integ-tests-alpha';
+
+const app = new App();
+const stack = new Stack(app, 'TestStack');
+
+// Create resources
+new s3.Bucket(stack, 'TestBucket');
+new dynamodb.Table(stack, 'TestTable', {
+  partitionKey: { name: 'id', type: dynamodb.AttributeType.STRING },
+});
+new iam.User(stack, 'TestUser');
+
+// Apply different removal policies to demonstrate functionality
+RemovalPolicies.of(stack).destroy();
```

Comment:
How about creating a custom construct that attaches a retain method? The following situation can be realized:

- Stack(: with destroy method)
  - Resource A: destroy
  - Resource B: destroy
  - RetainConstruct(: with retain method)
    - X: retain

Reply:
I looked at the code now; Bucket and Table default to RETAIN, so with the current stack, all resources are RETAIN. So it would be nice to have a resource that would be DESTROY.
Sorry for repeating.

Reply:
I have confirmed that the integ test reflecting this passes.

Reply:
With this code, the result would be the same even without `RemovalPolicies.of(stack).destroy();` and would not test that method.
So how about the following test?:

```ts
const app = new App();
const stack = new Stack(app, 'TestStack');

new Bucket(stack, 'TestBucket');

new Table(stack, 'TestTable', {
  partitionKey: { name: 'id', type: AttributeType.STRING },
});

new User(stack, 'TestUser');

const destroyBucket = new Construct(stack, 'DestroyBucket');
new Bucket(destroyBucket, 'Default');

RemovalPolicies.of(stack).retain({
  overwrite: true,
  priority: 50,
});
RemovalPolicies.of(destroyBucket).destroy({
  overwrite: true,
  priority: 100,
});
```

With this sample code, it would be good to just change the test code, since no changes are created in the snapshot. If you can delete the entire snapshot directory and re-run it, it would be good to have it written as you like, rather than as this code is.

Reply:
here: https://github.com/aws/aws-cdk/pull/32283#discussion_r1899110226

Reply:
here: https://github.com/aws/aws-cdk/pull/32283#discussion_r1899111637